src/spatial_reaction_systems/spatial_ODE_systems.jl

```diff
@@ -0,0 +1,182 @@
+            jac_prototype[get_index(e.src, s, lrs.num_species), get_index(e.src, s, lrs.num_species)] -= get_component_value(rates, e_idx)    # Term due to species leaving source vertex.
+            jac_prototype[get_index(e.src, s, lrs.num_species), get_index(e.dst, s, lrs.num_species)] += get_component_value(rates, e_idx)    # Term due to species arriving to destination vertex.
+        end
+    end
+
+    return jac_prototype
+end
+
+# Defines the forcing functor's effect on the (spatial) ODE system.
+function (f_func::LatticeDiffusionODEf)(du, u, p, t)
+    # Updates for non-spatial reactions.
+    for vert_i in 1:(f_func.num_verts)                                       # Loops through each vertex of the lattice. Applies the (non-spatial) ODEFunction to the species in that vertex.
+        f_func.ofunc((@view du[get_indexes(vert_i, f_func.num_species)]),    # Get the indexes of the i'th vertex (current one in the loop) in the u vector. Uses this to create a view of the vector to which the new species concentrations are written.  
+                     (@view u[get_indexes(vert_i, f_func.num_species)]),     # Same as above, but reads the current species concentrations.
+                      view_vert_ps_vector!(f_func.work_vert_ps, p, vert_i, f_func.enum_v_ps_idx_types),   # Gets a vector with the values of the (vertex) parameters in the current vertex.
+                      t)                                                     # Time.
+    end
+
+    # Updates for spatial reactions.
+    for (s_idx, (s, rates)) in enumerate(f_func.transport_rates)            # Loops through all species with transportation. Here: s_idx is its index among the species with transportations. s is its index among all species (in the species(::ReactionSystem) vector). rates is its rates values (vector length 1 if uniform, else same length as the number of edges).
+        for vert_i in 1:(f_func.num_verts)                                  # Loops through all vertexes.
+            du[get_index(vert_i, s, f_func.num_species)] -= f_func.leaving_rates[s_idx, vert_i] *
+                                                u[get_index(vert_i, s,
+                                                f_func.num_species)]        # Finds the leaving rate of this species in this vertex. Updates the du vector at that vertex/species combination with the corresponding rate (leaving rate times concentration).
+        end
+        for (e_idx, e) in f_func.enum_edges  # Loops through all edges.
+            du[get_index(e.dst, s, f_func.num_species)] += get_component_value(rates, e_idx) *
+                                                  u[get_index(e.src, s,
+                                                  f_func.num_species)]      # For the destination of this edge, we want to add the influx term to du. This is ["rates" value for this edge]*[the species concentration in the source vertex]. 
+        end
+    end
```

Comment:
```suggestion
    # s_idx is species index among transport species, s is index among all species
    # rates are the species' transport rates
    for (s_idx, (s, rates)) in enumerate(f_func.transport_rates)  
        # rate for leaving vert_i
        for vert_i in 1:(f_func.num_verts)                                 
            idx = get_index(vert_i, s, f_func.num_species)
            du[idx] -= f_func.leaving_rates[s_idx, vert_i] * u[idx]
        end
        # add rates for entering a given vertex via an incoming edge
        for (e_idx, e) in f_func.enum_edges 
            idx = get_index(e.dst, s, f_func.num_species)
            du[idx] += get_component_value(rates, e_idx) * u[idx] 
        end
    end
```
Your loop order seems flipped for performance. If `du` is ordered by all species at vertex one, all species at vertex 2, etc, I'd expect the outer loop to be over vertices and the inner loop over species to traverse the vector in memory order. The ordering you are using is what I'd expect if we order by species one at all locations, species 2 at all locations, etc. (Which we might want to switch to as I've mentioned as it is better for the transport operator handling.)